references/video_classification/train.py

```diff
@@ -203,6 +203,7 @@ def main(args):
 
     print("Creating model")
     model = torchvision.models.video.__dict__[args.model](pretrained=args.pretrained)
+    model.fc.out_features = args.output_classes
```

Comment:
This is not quite what I meant.

Changing `model.fc.out_features` doesn't change the number of output planes, because the [model weights have already been generated](https://github.com/pytorch/pytorch/blob/2460dced8fb9fa267dc33948cefe41f18d4b9a5e/torch/nn/modules/linear.py#L68-L77).
What I meant was to do something like
```python
model = torchvision.models.video.__dict__[args.model](pretrained=args.pretrained, num_classes=args.num_classes)
```
But then, using `pretrained=True` and num-classes different than 400 will give an error, because the state-dicts are not compatible.

I'd rather not expose this option to the training scripts, as the user can modify the model as they want to perform their type of fine-tuning.

Reply:
If fine-tuning on a custom dataset, one loads in a pretrained model with that model's `num_class` (400 here), and then changes the output classes to their custom dataset.

Though the output planes are different, the biggest benefit from a pretrained model aren't in the `fc` part of the model, but the convolutional filters that come in the previous parts of the model. The `fc` parts get retrained when fine-tuning this way. 

> I'd rather not expose this option to the training scripts, as the user can modify the model as they want to perform their type of fine-tuning.

This change in `model.fc.out_features` is mandatory when fine-tuning on a custom dataset, so I thought it would make sense to have it in the `args`.  But of course, that's your call :)

Reply:
What I mean is that this change doesn't actually change anything in the model. The weights are still the same as before, and they are not resized to a different size.

Putting it differently, you'll only predict at most 400 classes in this way.